Feature/be 6419 (#1)
appcircleio/appcircle-custom-script-from-git-component: README.md

# Appcircle Custom Script from Git Component

> Clone or use an existing Git repository, execute a specified custom script (bash, Python, Ruby, Node.js, Perl, or Java), and expose the workspace path via environment variables in your Appcircle workflow.

Reviewer: 1. The phrase “already exists” in the first sentence might not clearly convey that it refers to the repository being available on the local machine. Could you please rephrase it for clarity?
2. Please paying attention to capitalization would improve clarity and consistency
```suggestion
Clone a Git repository from a remote source or use a repository that already exists on the local machine, execute a specified custom script (Bash, Python, Ruby, Node.js, Perl, or Java), and expose the workspace path via environment variables in your Appcircle workflow.
```

## Required Inputs

- `AC_SCRIPT_FILENAME`: Name of the script file to execute (e.g., build.sh, code.rb, test.py). Which is contained within the Git repository. The script file has to be under the root folder of the Git repository.

## Optional Inputs

> One of the following input groups (either AC_SCRIPT_REPO_DIR or AC_SCRIPT_REPO_CLONE_URL, AC_SCRIPT_GIT_USERNAME, AC_SCRIPT_GIT_PAT) must be provided. If the repository is public, AC_SCRIPT_REPO_CLONE_URL is enough.

Reviewer: To improve readability and consistency, please wrap the environment variable names (like AC_SCRIPT_REPO_DIR, AC_SCRIPT_REPO_CLONE_URL, etc.) in backticks (``) so they are formatted as code.

> - `AC_SCRIPT_REPO_DIR`: Local directory path where your script already exists in the build machine for reuse in subsequent steps.

Reviewer: ```suggestion
- `AC_SCRIPT_REPO_DIR`: Local directory path of script if the script already exists in the runner for reuse in subsequent steps.
```

- `AC_SCRIPT_REPO_CLONE_URL`: Git clone URL of the repository containing your script.

- `AC_SCRIPT_GIT_USERNAME`: Username for private repository authentication. 
- `AC_SCRIPT_GIT_PAT`: Personal Access Token (PAT) for private repository authentication.

- `AC_SCRIPT_GIT_BRANCH`: Git branch to check out before execution (defaults to main).

> - `AC_SCRIPT_EXTRA_PARAMETERS`: Additional parameters to pass to the script (comma "," separated; if a parameter has an empty character, define it with " ").

Reviewer: Could you add an example?

## Output Variable

> - `AC_SCRIPT_REPO_OUTPUT_DIR`: To reuse the Git repository in subsequent steps, generate a local directory path key where your repository has been cloned in the build machine.

Reviewer: 1. Instead of "build machine," using "runner" would be more appropriate and align better with common terminology.
2. It might help to clarify what is meant by "workflow" to make the context clearer.

```suggestion
- `AC_SCRIPT_REPO_OUTPUT_DIR`: Specifies a local directory path where the Git repository is cloned on the runner. This allows reusing the same repository in subsequent steps of the workflow..
```

# Appcircle Custom Script from Git Component

Clone a Git repository from a remote source or use a repository that already exists on the local machine, execute a specified custom script (Bash, Python, Ruby, Node.js, Perl, or Java), and expose the workspace path via environment variables in your Appcircle workflow.

## Required Inputs

- `AC_SCRIPT_FILENAME`: Name of the script file to execute (e.g., `build.sh`, `code.rb`, `test.py`). Which is contained within the Git repository.

## Optional Inputs

> One of the following input groups (either `AC_SCRIPT_REPO_DIR` or `AC_SCRIPT_REPO_CLONE_URL`, `AC_SCRIPT_GIT_USERNAME`, `AC_SCRIPT_GIT_PAT`) must be provided. If the repository is public, `AC_SCRIPT_REPO_CLONE_URL` is enough.

Reviewer: The explanation here may be misleading, we should mention each obligation in its own ENV or explain it in detail here.

> - `AC_SCRIPT_REPO_DIR`: If the Git repository has already been cloned in a previous step of the same workflow, set the Repository Directory Output here. This input is required if `Git Clone URL` is not provided.

Reviewer: Could you please apply the changes you made in the component yaml here as well?

> - `AC_SCRIPT_REPO_CLONE_URL`: Git repository clone URL (https). Required if the Repository Directory Output is not provided. (e.g. https://exampleGit.exampleRepo.git).

Reviewer: The (https) statement is no longer necessary.

- `AC_SCRIPT_GIT_USERNAME`: Git provider username for authentication. This is required if the Repository Directory input is not provided. A public repository does not require a username.
- `AC_SCRIPT_GIT_PAT`: Git provider personal access token for authentication. Required if the `Repository Directory` input is not provided. A public repository does not require a PAT.

- `AC_SCRIPT_GIT_BRANCH`: Name of the branch to check out from the script repository. If not specified, the repository's default branch will be used.

> - `AC_SCRIPT_EXTRA_PARAMETERS`: Additional parameters to pass to the script (comma "," separated; if a parameter has an empty character, define it with " "; e.g. `param1,param2,"string with spaces",param3`).

Reviewer: ```suggestion
- `AC_SCRIPT_EXTRA_PARAMETERS`: Additional parameters to pass to the script (comma "," separated; if a parameter has an empty character, define it with " "; e.g. `param1,param2,"param3 with spaces",param4`).
```

## Required Inputs

- `AC_SCRIPT_FILENAME`: Specifies the exact name of the script file to execute within the Git repository (e.g., `test.sh`, `code.rb` or `relative_path/python.py`).

## Optional Inputs

- `AC_SCRIPT_REPO_DIR`: If the Git repository has already been cloned in a previous step of the same workflow, set the Repository Directory Output here. This input is required if `AC_SCRIPT_REPO_CLONE_URL` is not provided.
- `AC_SCRIPT_REPO_CLONE_URL`: Git repository clone URL. Required if the Repository Directory Output is not provided. (e.g. https://exampleGit.exampleRepo.git). This input is required if `AC_SCRIPT_REPO_DIR` is not provided.

> - `AC_SCRIPT_GIT_USERNAME`: Git provider username for authentication. This is required if the `AC_SCRIPT_REPO_DIR` input is not provided. A public repository does not require a username.

Reviewer: ```suggestion
- `AC_SCRIPT_GIT_USERNAME`: Git provider username for authentication. This is required if the `AC_SCRIPT_REPO_DIR` input is not provided and the Git repository is private.
```

> - `AC_SCRIPT_GIT_PAT`: Git provider personal access token for authentication. This is required if the `AC_SCRIPT_REPO_DIR` input is not provided. A public repository does not require a PAT.

Reviewer: ```suggestion
- `AC_SCRIPT_GIT_PAT`: Git provider personal access token for authentication. This is required if the `AC_SCRIPT_REPO_DIR` input is not provided and the Git repository is private.
```
